File: src/lib/api/openclaw.ts
import { invoke } from "./adapter";

export interface OpenClawDefaultModel {
  primary: string;
  fallbacks: string[];
}

export interface OpenClawHealthWarning {
  code: string;
  message: string;
  path?: string;
}

export interface OpenClawLiveModelSummary {
  id: string;
  name?: string;
  alias?: string;
}

export interface OpenClawLiveProviderSummary {
  id: string;
  baseUrl?: string;
  api?: string;
  models: OpenClawLiveModelSummary[];
  hasApiKey: boolean;
}

export interface OpenClawLiveStatus {
  defaultModel?: OpenClawDefaultModel;
  providers: OpenClawLiveProviderSummary[];
  warnings: OpenClawHealthWarning[];
}

export interface OpenClawWriteOutcome {
  backupPath?: string;
  warnings: OpenClawHealthWarning[];
}

export const openclawApi = {
  getStatus(): Promise<OpenClawLiveStatus> {
    return invoke("get_openclaw_status");
  },
  getProviders(): Promise<OpenClawLiveProviderSummary[]> {
    return invoke("get_openclaw_live_providers");
  },
  getProvider(providerId: string): Promise<OpenClawLiveProviderSummary | null> {
    return invoke("get_openclaw_live_provider", { providerId });
  },
  getDefaultModel(): Promise<OpenClawDefaultModel | null> {
    return invoke("get_openclaw_default_model");
  },
  setDefaultModel(model: OpenClawDefaultModel): Promise<OpenClawWriteOutcome> {
    return invoke("set_openclaw_default_model", { model });
  },
  clearDefaultModel(): Promise<OpenClawWriteOutcome> {
    return invoke("clear_openclaw_default_model");
  },
  getHealth(): Promise<OpenClawHealthWarning[]> {
    return invoke("scan_openclaw_config_health");
  },
};

File: src-tauri/src/commands/openclaw.rs
use crate::openclaw_config::{
    self, OpenClawDefaultModel, OpenClawHealthWarning, OpenClawLiveProviderSummary,
    OpenClawLiveStatus, OpenClawWriteOutcome,
};

#[tauri::command]
pub async fn get_openclaw_status() -> Result<OpenClawLiveStatus, String> {
    openclaw_config::get_live_status().map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn get_openclaw_live_providers() -> Result<Vec<OpenClawLiveProviderSummary>, String> {
    openclaw_config::get_live_provider_summaries().map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn get_openclaw_live_provider(
    provider_id: String,
) -> Result<Option<OpenClawLiveProviderSummary>, String> {
    openclaw_config::get_live_provider_summary(&provider_id).map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn get_openclaw_default_model() -> Result<Option<OpenClawDefaultModel>, String> {
    openclaw_config::get_default_model().map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn set_openclaw_default_model(
    model: OpenClawDefaultModel,
) -> Result<OpenClawWriteOutcome, String> {
    openclaw_config::set_default_model(&model).map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn clear_openclaw_default_model() -> Result<OpenClawWriteOutcome, String> {
    openclaw_config::clear_default_model().map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn scan_openclaw_config_health() -> Result<Vec<OpenClawHealthWarning>, String> {
    openclaw_config::scan_openclaw_config_health().map_err(|error| error.to_string())
}
